modules/telemetry/json_packets.py

```diff
@@ -185,3 +188,167 @@ def parse_mission_file(mission_file: Path) -> MissionEntry:
+        for data_packet in output_specification.keys():
+            for stored_value in output_specification[data_packet].keys():
+                for version in output_specification[data_packet][stored_value].keys():
+                    for block in output_specification[data_packet][stored_value][version].keys():
+                        input_key: str = output_specification[data_packet][stored_value][version][block]
+                        output_key: str = f"{data_packet}.{stored_value}"
+
+                        existing: dict[str, str] = self.decoder[int(version)].get(int(block), {})
+                        existing[input_key] = output_key
+                        self.decoder[int(version)][int(block)] = existing
+
+    def updateTelemetry(self, packet_version: int, blocks: list[ParsedBlock]) -> None:
+        """Updates telemetry object from given parsed blocks"""
+
+        # Extract block data
+        for block in blocks:
+            block_num: int = block.block_header.message_subtype
+            block_decode: dict[str, str] = self.decoder[packet_version][block_num]
+            data: dict[str, int | dict[str, int]] = block.block_contents
+
+            logger.debug(f"{block}")
+
+            try:
+                # Update last mission time
+                if data["mission_time"] > self.last_mission_time:  # type: ignore
+                    self.last_mission_time = data["mission_time"]  # type: ignore
+
+                # Grab input values and put them in update buffer (to fill output packets)
+                for key in block_decode.keys():
+                    destinationBlock: str = block_decode[key].split(".")[0]
+                    destinationValue: str = block_decode[key].split(".")[1]
+                    # Extract data and associated mission time to buffer
+                    self.update_buffer[destinationBlock]["mission_time"] = data["mission_time"]  # type: ignore
+                    if "." in key:
+                        datakey: str = key.split(".")[0]
+                        datasubkey: str = key.split(".")[1]
+                        self.update_buffer[destinationBlock][destinationValue] = data[datakey][  # type: ignore
+                            datasubkey
+                        ]
+                    else:
+                        self.update_buffer[destinationBlock][destinationValue] = data[key]  # type: ignore
+
+                # Check if we filled any packet during this block extraction
+                for key in self.update_buffer.keys():
+                    if None not in self.update_buffer[key].values():
+                        # Let's update packet
+                        self.output_blocks[key].update(self.update_buffer[key], self.buffer_size)  # type: ignore
+                        # Clear packets buffer
+                        for subkey in self.update_buffer[key].keys():
+                            self.update_buffer[key][subkey] = None
+            except KeyError as e:
```

Comment:
I think it might be beneficial to crash here. Without the schema we won't be able to output telemetry, no?

Reply:
Upon testing this case only seems to gets thrown if the telemetry_packet.json and __iter__ outputs from that corresponding block in data_block gets thrown out of sync. I.e, if you put the metres variable as "altitude.met**re**s" but its actually "altitude.met**er**s" then it can't find the variable and will trigger. It's more of a sanity/verification when updating it. Additionally, I don't like the premise of crashing the whole software if a single variable is missing.